app/scripts/views/mixins/avatar-mixin.js

```diff
@@ -81,6 +110,39 @@ define(function (require, exports, module) {
       return ! account.has('profileImageUrl');
     },
 
+    _addLoadingSpinner: function (spinnerWrapperEl) {
+      return $('<span class="avatar-spinner"></span>').appendTo(spinnerWrapperEl.addClass('with-spinner'));
+    },
+
+    // "Completes" the spinner, transitioning the semi-circle to a circle, and
+    // then removes the spinner element.
+    _completeLoadingSpinner: function (spinnerEl, timeout) {
+      var deferred = p.defer();
+      spinnerEl
+        .addClass('completed')
+        .on('transitionend', function (event) {
+          // The first transitionend event will resolve the promise, but the spinner will have
+          // subsequent transitions, so we'll also hook on the transitionend event of the
+          // ::after pseudoelement, which "expands" to hide the spinner.
+          deferred.resolve();
+
+          if (event.originalEvent && event.originalEvent.pseudoElement === '::after') {
+            spinnerEl.remove();
+          }
+        });
+
+
+      // Always resolve and remove the spinner after MAX_SPINNER_COMPLETE_TIME,
+      // in case we don't receive the expected transitionend events, such as in
+      // the case of IE.
+      setTimeout(function transitionMaxTime () {
```

Comment:
Can you use a `this.setTimeout` here? Our views handle [stopping any outstanding timeouts](https://github.com/mozilla/fxa-content-server/blob/84206c73c519f9f0d5c971f2bd3fc9ba9aa0f664/app/scripts/views/mixins/timer-mixin.js#L22-L23) when the view is destroyed. Calling `view.setTimeout` also allows `setTimeout` [to be stubbed for unit tests](https://github.com/mozilla/fxa-content-server/blob/84206c73c519f9f0d5c971f2bd3fc9ba9aa0f664/app/tests/spec/views/confirm.js#L175) to call the callback immediately, allowing the tests to run a tad bit more quickly.